src/Logary/DataModel.fs

```diff
@@ -302,11 +285,13 @@ type Logger =
   /// `Messages` produced from this instance.
   abstract name: PointName
 
-  /// - `waitForBuffers`: This causes the logger/caller to block on the RingBuffer being
-  ///   available to take the message. Giving a true here may cause the
-  abstract logWithAck: waitForBuffers:bool * level:LogLevel -> messageFactory:(LogLevel -> Message) -> LogResult
-
-  //abstract log: Message -> Alt<Result<Promise<unit>, unit>>
+  /// Returns an Alt that commits on ALL N Targets' buffers accepting the message.
+  /// Even if the Alt was not committed to, one or more targets (fewer than N) may have accepted the message.
+  /// And this can not be canceled.
+  ///
+  /// - `putBufferTimeOut`: time out duration waiting for each target buffer to be available to take the message.
```

Comment:
I don't think we should have a buffer timeout at all; if it can't take the message, we must reject the message (optionally logging that internally!)

Reply:
if the timeout is `zero`, it act like the old `trylog`, it will return a timeout error immediately, and the registy's `logResultHandler` can notify this situation.

Reply:
https://github.com/logary/logary/blob/6b049fa05534bb55b172c237f09e2f16eec5c851/src/Logary/TargetModule.fs#L42

it has been considered. see here